testutil-server/src/main/java/io/spine/testing/server/blackbox/MultitenantBlackBoxContext.java

```diff
@@ -0,0 +1,118 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.testing.server.blackbox;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.spine.core.Command;
+import io.spine.core.Commands;
+import io.spine.core.TenantId;
+import io.spine.server.event.Enricher;
+import io.spine.server.tenant.TenantAwareRunner;
+import io.spine.testing.client.TestActorRequestFactory;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * A black box bounded context for writing integration tests in a multitenant environment.
+ */
+@VisibleForTesting
+public class MultitenantBlackBoxContext
+        extends BlackBoxBoundedContext<MultitenantBlackBoxContext> {
+
+    private TenantId tenantId;
+
+    /**
+     * Creates a new multi-tenant instance.
+     */
+    MultitenantBlackBoxContext(Enricher enricher) {
+        super(true, enricher);
+    }
+
+    /**
+     * Switches the bounded context to operate on behalf of the specified tenant.
+     *
+     * @param tenant
+     *         new tenant ID
+     * @return current instance
+     */
+    public MultitenantBlackBoxContext withTenant(TenantId tenant) {
+        this.tenantId = tenant;
+        return this;
+    }
+
+    @Override
+    protected TestActorRequestFactory requestFactory() {
+        return requestFactory(tenantId());
+    }
+
+    @Override
+    protected EmittedCommands emittedCommands(CommandMemoizingTap commandTap) {
+        List<Command> allCommands = commandTap.commands();
+        List<Command> tenantCommands = allCommands.stream()
+                                                  .filter(new IsTenantCommand(tenantId))
+                                                  .collect(Collectors.toList());
+        return new EmittedCommands(tenantCommands);
+    }
+
+    @Override
+    protected EmittedEvents emittedEvents() {
+        TenantAwareRunner tenantAwareRunner = TenantAwareRunner.with(tenantId);
+        EmittedEvents events = tenantAwareRunner.evaluate(super::emittedEvents);
+        return events;
+    }
+
+    private TenantId tenantId() {
+        checkState(tenantId != null,
+                   "Set a tenant ID before calling receive and assert methods");
+        return tenantId;
+    }
+
+    /**
+     * Creates a new {@link io.spine.client.ActorRequestFactory actor request factory} for tests
+     * with a provided tenant ID.
+     *
+     * @param tenantId
+     *         an identifier of a tenant that is executing requests in this Bounded Context
+     * @return a new request factory instance
+     */
+    private static TestActorRequestFactory requestFactory(TenantId tenantId) {
+        return TestActorRequestFactory.newInstance(MultitenantBlackBoxContext.class, tenantId);
+    }
+
+    private static class IsTenantCommand implements Predicate<Command> {
```

Comment:
Please document. And I think when you do that, you will have to rename this class to match the description.